[PD-7809] map markers

nate998877 opened the pull request:
## What does this PR do?
Merges the 4 map components that previously existed into one. With the goal of creating one place to add bug fixes & extend functionality. The existing military & job maps have diverged from each other regarding features & functionality. 


## Notes to Reviewer
Testing/ examples available at https://github.com/DirectEmployers/microsite-template/tree/map-example

Please ensure `NUXT_ENV_GM_API_KEY` is set inside .env.development in the template. Otherwise you will receive 500 errors

surgiie commented on src/components/Search/Maps/AppMap.vue:
```diff
@@ -0,0 +1,455 @@
+        },
+        assignIcons(marker){
+            // marker.icon = `https://d12wqovxet6953.cloudfront.net/jobmap/images/${marker.company_exact.toLowerCase()}.png`
+            marker.icon = "https://frontdoor.jobs/favicon.ico"
```
This shouldnt be hard coded to a specific company.

surgiie reviewed (changes requested):
Minor code changes noted. Will re-review when PR/example is ready.

surgiie commented on src/components/Search/Maps/AppMilitaryMap.vue:
```diff
@@ -0,0 +1,70 @@
+}
+</script>
+
+{
```
Dont think you meant to leave this here?

surgiie reviewed (changes requested):
* Clicking on a job map marker in your template example doesnt navigate to search page
* Im a bit hesistant to merge because of the use of "Extends" im under the impression this is not available or discouraged in vue 3.
* Your military example is also broken so unable to test that map:

![image](https://user-images.githubusercontent.com/12025002/162018963-afddeb46-05f8-4a4d-b48a-394ca077d83a.png)

* I would also prefer that @zarifrahat plays around with these examples, particularly the military one once its working since he is using it on sites currently.

nate998877 commented:
> * Clicking on a job map marker in your template example doesnt navigate to search page
> * Im a bit hesistant to merge because of the use of "Extends" im under the impression this is not available or discouraged in vue 3.
> * Your military example is also broken so unable to test that map:
> * I would also prefer that @zarifrahat plays around with these examples, particularly the military one once its working since he is using it on sites currently.

All of the maps work, it appears you maybe were missing an API key, I updated the comments to reflect that. I thought we all had API keys already set.

The job map has been updated to fix the navigation issues.

Zarif defiantly needs to play with these & I have added him as a reviewer.
